Mathlib/Algebra/Algebra/Prod.lean

```diff
@@ -108,3 +109,23 @@ def prodEquiv : (A →ₐ[R] B) × (A →ₐ[R] C) ≃ (A →ₐ[R] B × C)
 #align alg_hom.prod_equiv AlgHom.prodEquiv
 
 end AlgHom
+
+section Unique
+
+/-- Multiplying by the trivial algebra from the left does not change the structure.
+
+This is the `AlgebraEquiv` version of `AddEquiv.uniqueProd`. -/
```

Comment:
Let's link the strongest weaker equivalence:
```suggestion
This is the `AlgebraEquiv` version of `LinearEquiv.uniqueProd`. -/
```
Ideally we'd mention`RingEquiv.uniqueProd` too, but I assume that's also missing.

Reply:
[RingEquiv.prodZeroRing](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Prod.html#RingEquiv.prodZeroRing) and [RingEquiv.zeroRingProd](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Prod.html#RingEquiv.zeroRingProd) exist, but they use the condition `[Subsingleton S]` rather than `[Unique S]` instead. Any thoughts on whether we'd like these to be more consistent with one another? In any case, probably worth referencing them to one another for ease of use.

Reply:
The naming discrepancy sounds like a question for Zulip.
Using `Subsingleton` instead of `Unique` does seem sensible here.

Reply:
(I'm happy to merge as is with a reference to the inconsistent ring name, if you'd prefer to do this in two steps)

Reply:
Opened a Zulip thread about that [here.](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/prodZeroRing.20vs.20prodUnique) I'll already commit a change to make them refer to one another tho